File: Ahorn/triggers/bgModeTrigger.jl
module BGswitchBGModeTrigger

using ..Ahorn, Maple

@mapdef Trigger "bgSwitch/bgTrigger" BGModeTrigger(x::Integer, y::Integer, width::Integer=Maple.defaultTriggerWidth, height::Integer=Maple.defaultTriggerHeight,
    mode::Bool=true, persistent::Bool=false, playEffects::Bool=true) 

const placements = Ahorn.PlacementDict(
    "BG Mode Trigger (On, BGswitch)" => Ahorn.EntityPlacement(
        BGModeTrigger,
        "rectangle",
        Dict{String, Any}(
            "mode" => true,
            "persistent" => false,
            "playEffects" => true
        )
    ),
    "BG Mode Trigger (Off, BGswitch)" => Ahorn.EntityPlacement(
        BGModeTrigger,
        "rectangle",
        Dict{String, Any}(
            "mode" => false,
            "persistent" => false,
            "playEffects" => true
        )
    )
)

end

File: Loenn/triggers/bg_mode_trigger.lua
local bgModeTrigger = {}

bgModeTrigger.name = "bgSwitch/bgTrigger"
bgModeTrigger.placements = {
    {
        name = "on",
        data = {
            mode = true,
            persistent = false,
            playEffects = true
        }
    },
    {
        name = "off",
        data = {
            mode = false,
            persistent = false,
            playEffects = true
        }
    }
}

return bgModeTrigger